src/sentry/models/projectkey.py

```diff
@@ -89,13 +99,16 @@ class roles(TypedClassBitField):
 
     objects: ClassVar[ProjectKeyManager] = ProjectKeyManager(
         cache_fields=("public_key", "secret_key"),
-        # store projectkeys in memcached for longer than other models,
-        # specifically to make the relay_projectconfig endpoint faster.
-        cache_ttl=60 * 30,
     )
 
     data: models.Field[dict[str, Any], dict[str, Any]] = JSONField()
 
+    use_case = models.CharField(
+        max_length=32,
+        choices=[(v.value, v.value) for v in UseCase],
+        default=UseCase.USER.value,
+    )
```

Comment:
I considered encoding the use case into the existing `roles` field. But `roles` specify the permissions a DSN has, while the use case defines who may use the DSN.